src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -19,255 +18,99 @@ namespace Microsoft.CodeAnalysis.PatternMatching
 
-        private static readonly char[] s_dotCharacterArray = { '.' };
-
         private readonly object _gate = new object();
 
+        private readonly bool _includeMatchedSpans;
         private readonly bool _allowFuzzyMatching;
-        private readonly bool _invalidPattern;
-        private readonly PatternSegment _fullPatternSegment;
-        private readonly PatternSegment[] _dotSeparatedPatternSegments;
 
         private readonly Dictionary<string, StringBreaks> _stringToWordSpans = new Dictionary<string, StringBreaks>();
         private static readonly Func<string, StringBreaks> _breakIntoWordSpans = StringBreaker.BreakIntoWordParts;
 
         // PERF: Cache the culture's compareInfo to avoid the overhead of asking for them repeatedly in inner loops
         private readonly CompareInfo _compareInfo;
 
+        private bool _invalidPattern;
         /// <summary>
         /// Construct a new PatternMatcher using the specified culture.
         /// </summary>
-        /// <param name="pattern">The pattern to make the pattern matcher for.</param>
         /// <param name="culture">The culture to use for string searching and comparison.</param>
+        /// <param name="includeMatchedSpans">Whether or not the matching parts of the candidate should be supplied in results.</param>
         /// <param name="allowFuzzyMatching">Whether or not close matches should count as matches.</param>
-        public PatternMatcher(
-            string pattern,
-            CultureInfo culture = null,
+        protected PatternMatcher(
+            bool includeMatchedSpans,
+            CultureInfo culture,
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
-            pattern = pattern.Trim();
             _compareInfo = culture.CompareInfo;
+            _includeMatchedSpans = includeMatchedSpans;
             _allowFuzzyMatching = allowFuzzyMatching;
-
-            _fullPatternSegment = new PatternSegment(pattern, allowFuzzyMatching);
-
-            if (pattern.IndexOf('.') < 0)
-            {
-                // PERF: Avoid string.Split allocations when the pattern doesn't contain a dot.
-                _dotSeparatedPatternSegments = pattern.Length > 0
-                    ? new PatternSegment[1] { _fullPatternSegment }
-                    : Array.Empty<PatternSegment>();
-            }
-            else
-            {
-                _dotSeparatedPatternSegments = pattern.Split(s_dotCharacterArray, StringSplitOptions.RemoveEmptyEntries)
-                                                .Select(text => new PatternSegment(text.Trim(), allowFuzzyMatching))
-                                                .ToArray();
-            }
-
-            _invalidPattern = _dotSeparatedPatternSegments.Length == 0 || _dotSeparatedPatternSegments.Any(s => s.IsInvalid);
         }
 
-        public void Dispose()
+        public virtual void Dispose()
```

Comment:
❗️ Use correct `Dispose()` pattern. The previous implementation was only allowed to be in `Dispose()` (as opposed to `Dispose(bool)`) because the type was sealed.